### PR TITLE
[NFI]: Remove NVIDIA specific code

### PR DESCRIPTION
**Pull request by etiotto**

### CODE REVIEW

File: third_party/intel/lib/TritonIntelGPUToLLVM/ControlFlowOpToLLVM.cpp [hunk 1]
```diff
@@ -16,6 +16,7 @@ struct ReturnOpConversion
                   ConversionPatternRewriter &rewriter) const override {
     auto funcOp = op->getParentOfType<LLVM::LLVMFuncOp>();
     if (funcOp->hasAttr("nvvm.kernel")) {
+      assert(false && "On Intel this code should be unreachable");
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/ControlFlowOpToLLVM.cpp:19]: change to llvm_unreachable

File: third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp [hunk 2]
```diff
@@ -411,6 +411,7 @@ void createBarrier(ConversionPatternRewriter &rewriter, Location loc,
   if (numCTAs == 1) {
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp:411]: assert(numCTAs == 1 && ....);

File: third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp [hunk 3]
```diff
@@ -411,6 +411,7 @@ void createBarrier(ConversionPatternRewriter &rewriter, Location loc,
...
     barrier();
   } else {
+    assert(false && "Cannot use NVIDIA_GPU operations");
     rewriter.create<triton::nvidia_gpu::ClusterArriveOp>(loc, false);
     rewriter.create<triton::nvidia_gpu::ClusterWaitOp>(loc);
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp:416]: remove?

File: third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp (hunk 2 above)
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp]: done

File: third_party/intel/lib/TritonIntelGPUToLLVM/ControlFlowOpToLLVM.cpp (hunk 1 above)
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ControlFlowOpToLLVM.cpp]: done

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**